lib/carbon/cache.py

```diff
@@ -116,12 +119,28 @@ def counts(self):
     return [(metric, len(datapoints)) for (metric, datapoints) in self.items()]
 
   @property
+  def unflush_counts(self):
+    return [(metric, self._count_unflushed(datapoints)) for (metric, datapoints) in self.items()]
```

Comment:
Maintaining a large map {"metric": unflushed_count } ? i can do that as well.
 As we are introducing the "is_flushed" field, we are going to pay the cost anyway.